File: backend/src/main/java/com/HTPj/htpj/service/AddonServiceService.java
package com.HTPj.htpj.service;

import com.HTPj.htpj.dto.request.addonservice.AddBookingAddonsRequest;
import com.HTPj.htpj.dto.request.addonservice.CreateAddonServiceRequest;
import com.HTPj.htpj.dto.request.addonservice.UpdateAddonServiceRequest;
import com.HTPj.htpj.dto.response.addonservice.AddonServiceResponse;
import com.HTPj.htpj.dto.response.addonservice.BookingAddonServiceResponse;

import java.util.List;

public interface AddonServiceService {

    // UC-066: Tạo dịch vụ bổ trợ
    AddonServiceResponse createService(CreateAddonServiceRequest request);

    // UC-067: Cập nhật dịch vụ bổ trợ
    AddonServiceResponse updateService(Long serviceId, UpdateAddonServiceRequest request);

    // UC-068: Xóa dịch vụ bổ trợ (đặt inactive)
    AddonServiceResponse deleteService(Long serviceId);

    // Toggle trạng thái active/inactive
    AddonServiceResponse toggleStatus(Long serviceId);

    // Xem danh sách dịch vụ theo khách sạn
    List<AddonServiceResponse> getServicesByHotelId(Integer hotelId, String category);

    // Xem dịch vụ active theo khách sạn (Agency dùng khi đặt phòng)
    List<AddonServiceResponse> getActiveServicesByHotelId(Integer hotelId);

    // UC-026: Thêm dịch vụ vào booking
    List<BookingAddonServiceResponse> addServicesToBooking(AddBookingAddonsRequest request);

    // Xem dịch vụ đã đặt trong một booking
    List<BookingAddonServiceResponse> getBookingAddonServices(Long bookingId);
}

File: frontend/src/services/addonService.service.js
import api from "./axios.config.js";

// 1. Tạo dịch vụ bổ trợ (UC-066)
const createAddonService = async (payload) => {
    const response = await api.post(`/addon-services`, payload);
    return response.data;
};

// 2. Cập nhật dịch vụ bổ trợ (UC-067)
const updateAddonService = async (serviceId, payload) => {
    const response = await api.put(`/addon-services/${serviceId}`, payload);
    return response.data;
};

// 3. Xóa dịch vụ bổ trợ (UC-068)
const deleteAddonService = async (serviceId) => {
    const response = await api.delete(`/addon-services/${serviceId}`);
    return response.data;
};

// 4. Toggle trạng thái dịch vụ
const toggleAddonServiceStatus = async (serviceId) => {
    const response = await api.patch(`/addon-services/${serviceId}/status`);
    return response.data;
};

// 5. Lấy danh sách dịch vụ theo hotel (Hotel admin)
const getAddonServicesByHotel = async (hotelId, category = null) => {
    const params = { hotelId };
    if (category) params.category = category;
    const response = await api.get(`/addon-services`, { params });
    return response.data;
};

// 6. Lấy dịch vụ active theo hotel (Agency khi checkout)
const getActiveAddonServicesByHotel = async (hotelId) => {
    const response = await api.get(`/addon-services/active`, { params: { hotelId } });
    return response.data;
};

// 7. UC-026: Thêm dịch vụ vào booking
const addServicesToBooking = async (payload) => {
    const response = await api.post(`/addon-services/booking`, payload);
    return response.data;
};

// 8. Lấy dịch vụ đã đặt trong booking
const getBookingAddonServices = async (bookingId) => {
    const response = await api.get(`/addon-services/booking/${bookingId}`);
    return response.data;
};

export const addonServiceApi = {
    createAddonService,
    updateAddonService,
    deleteAddonService,
    toggleAddonServiceStatus,
    getAddonServicesByHotel,
    getActiveAddonServicesByHotel,
    addServicesToBooking,
    getBookingAddonServices,
};
